lib/token.js

```diff
@@ -12,8 +12,37 @@ function getWellKnown(sdk) {
   return http.get(sdk, sdk.options.url + '/.well-known/openid-configuration');
 }
 
-function validateClaims(sdk, claims, aud) {
-  var iss = sdk.options.url;
+function getAuthServerUrl(sdk, authServer) {
+  authServer = authServer || sdk.options.authServer;
+  if (authServer) {
+    if (/^https?:/.test(authServer)) {
+      return authServer;
+    }
+    return sdk.options.url + '/oauth2/' + authServer;
+  }
+}
+
+var isAuthorizationServer = new RegExp('^https?://.*?/oauth2/.+');
```

Comment:
Since this stuff is kind of confusing, let's add comments and examples in the code to express why we're doing these things. I.e. in this case the idea that if they don't have a custom authServer, it won't have the oauth2 so we have to add it, vs. the oauth2 being embedded otherwise.